apps/ledger-live-mobile/src/screens/Swap/Form/index.tsx

```diff
@@ -148,13 +151,58 @@ export function SwapForm({
   });
 
   const exchangeRatesState = swapTransaction.swap?.rates;
+  const { partnersList, exchangeRateList } = useMemo(() => {
+    const partnerAndExchangeRateDefault = {
+      partnersList: [],
+      exchangeRateList: [],
+    };
+
+    return (
+      exchangeRatesState.value?.reduce<{
+        partnersList: string[];
+        exchangeRateList: string[];
+      }>((prev, curr) => {
+        return {
+          partnersList: [...prev.partnersList, curr.provider],
+          exchangeRateList: [
+            ...prev.exchangeRateList,
+            formatCurrencyUnit(getFeesUnit(swapTransaction.swap.to.currency!), curr.toAmount),
+          ],
+        };
+      }, partnerAndExchangeRateDefault) ?? partnerAndExchangeRateDefault
+    );
+  }, [exchangeRatesState.value, swapTransaction.swap.to.currency]);
+
   const swapError = swapTransaction.fromAmountError || exchangeRatesState?.error;
   const pageState = usePageState(swapTransaction, swapError);
   const swapKYC = useSelector(swapKYCSelector);
   const provider = exchangeRate?.provider;
   const providerKYC = provider ? swapKYC?.[provider] : undefined;
   const kycStatus = providerKYC?.status as ValidKYCStatus | "rejected";
 
+  const editRatesTrackingProps = JSON.stringify({
+    ...sharedSwapTracking,
+    provider,
+    partnersList,
+    exchangeRateList,
+    sourceCurrency: swapTransaction.swap.from.currency?.id,
+    targetCurrency: swapTransaction.swap.from.currency?.id,
+  });
+
+  useEffect(() => {
+    if (pageState === "loaded") {
+      track("Swap Form - Edit Rates", {
+        ...sharedSwapTracking,
+        provider,
+        partnersList,
+        exchangeRateList,
+        sourceCurrency: swapTransaction.swap.from.currency?.id,
+        targetCurrency: swapTransaction.swap.from.currency?.id,
+      });
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [pageState, editRatesTrackingProps]);
```

Comment:
What is the benefit of stringify the track object vs add here directly dependencies?

Reply:
By stringifying it here we are guaranteeing that the useEffect will only be rerun if there is a change in the editRatesTrackingProps. If we were to pass an object here the useEffect would re-run on each new creation of that object even if all the keys and values are the same.

Reply:
Are you sure if it running twice because of this and not because we are in dev mode?